WordPress/gutenberg
# Gallery Block: Use styled scrollbars for image captions

Fixes: #56201

## What?

This PR displays a styled scrollbar on the scrollbar that appears when an image in a gallery block has a long caption.

This will improve the appearance on the front end, especially for users using Windows or users who use browsers that display scrollbars by default.

Note: The screenshot below was taken with Chrome browser on Windows 11 OS.

### Before

![before](https://github.com/WordPress/gutenberg/assets/54422211/68de5302-0220-4af1-ab72-fa5621fdb1a6)

### After

![after](https://github.com/WordPress/gutenberg/assets/54422211/fbfe8762-2ff8-4007-bb11-c95b312c11aa)

## Why?
`overflow:auto` and `max-height: 60%` are applied to the caption area of the image in the Gallery Block. Therefore, if the caption text is long, this caption area will be scrollable.

This has a impact on appearance on some OS and browsers. For example, when using Chrome on Windows OS, scroll bars are always displayed.

## How?
I applied the `custom-scrollbars-on-hover` mixin to this caption, which styles the scrollbar and makes it appear only on hover. This mixin is primarily used in the sidebar of the site editor. Therefore, I'm a little unsure whether this mixin can be used on the front end as well. I would like to hear everyone's opinions.

![sidebar](https://github.com/WordPress/gutenberg/assets/54422211/a4ad5048-7426-45f9-a2b5-b64241ef9714)

## Testing Instructions

- Insert a Gallery block and insert some Image blocks inside.
- Enter a long caption in one Image block.
- Check the scrollbar style.
- The scrollbar should only appear when you hover over it.
- Test in multiple browsers and multiple themes.
